e2e/scripts/aws-s3-client.test.js

```diff
@@ -0,0 +1,336 @@
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+const {S3Client, PutObjectCommand, GetObjectCommand} = require('@aws-sdk/client-s3')
+const {STSClient, AssumeRoleCommand} = require('@aws-sdk/client-sts')
+const SecureS3Client = require('./aws-s3-client')
+const {PWA_KIT_BOT_USER_SESSION, AWS_ACCESS_READ_WRITE, AWS_DEFAULT_REGION} = require('./constants')
+
+// Mock AWS SDK modules
+jest.mock('@aws-sdk/client-s3')
+jest.mock('@aws-sdk/client-sts')
+
+// Mock console methods to avoid cluttering test output
+const originalConsoleLog = console.log
+const originalConsoleError = console.error
+
+describe('SecureS3Client', () => {
+    let client
+    let mockS3Client
+    let mockSTSClient
+
+    beforeEach(() => {
+        // Reset mocks
+        jest.clearAllMocks()
+
+        // Mock console methods
+        console.log = jest.fn()
+        console.error = jest.fn()
+
+        // Setup mock S3 client
+        mockS3Client = {
+            send: jest.fn()
+        }
+        S3Client.mockImplementation(() => mockS3Client)
+
+        // Setup mock STS client
+        mockSTSClient = {
+            send: jest.fn()
+        }
+        STSClient.mockImplementation(() => mockSTSClient)
+    })
+
+    afterEach(() => {
+        // Restore console methods
+        console.log = originalConsoleLog
+        console.error = originalConsoleError
+    })
+
+    describe('constructor', () => {
+        it('should create instance with default options', () => {
+            client = new SecureS3Client()
+
+            expect(client.roleArn).toBeUndefined()
+            expect(client.roleSessionName).toBe(PWA_KIT_BOT_USER_SESSION)
+            expect(client.region).toBe(AWS_DEFAULT_REGION)
+            expect(client.readOnly).toBeUndefined()
+            expect(client.externalId).toBeUndefined()
+            expect(client.credentials).toBeNull()
+        })
+
+        it('should create instance with custom options', () => {
+            const options = {
+                roleArn: 'arn:aws:iam::123456789012:role/test-role',
+                roleSessionName: 'test-session',
+                region: 'us-west-2',
+                readOnly: true,
+                externalId: 'test-external-id'
+            }
+
+            client = new SecureS3Client(options)
+
+            expect(client.roleArn).toBe(options.roleArn)
+            expect(client.roleSessionName).toBe(options.roleSessionName)
+            expect(client.region).toBe(options.region)
+            expect(client.readOnly).toBe(options.readOnly)
+            expect(client.externalId).toBe(options.externalId)
+        })
+    })
+
+    describe('initialize', () => {
+        beforeEach(() => {
+            client = new SecureS3Client()
+        })
+
+        it('should initialize without role assumption when roleArn is not provided', async () => {
```

Comment:
Nit: Let's use test('') for constency and recoverability in IDE

Reply:
Updated 👍